adjusted architecture documentation to improve readability

dancesWithCycles opened the pull request:
Hi folks,
Thank you very much for creating and maintaining this repository and please accept my appreciation.

This PR shall improve the readability of the file [ARCHETECTURE.md](https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/ARCHITECTURE.md) by adjusting grammar and mode of expression.

I do not have adjusted the change log yet as I do not know the procedure for documentation.

Please feel free to checkout and merge this PR.

Cheers!

flaktack reviewed (changes requested):
Thanks for taking the time to improve the documentation @dancesWithCycles.

The paragraphs in the markdown documentation are formatted to 100 character lines, but your changes seem to have one-line-per-paragraph. Could you please reflow the paragraphs in your text editor to follow the existing convention of 100 character lines?

Reflowing the text will also make it easier to see what the actual changes are, since currently the diff contains no indication of what words were changed.

New changes to `ARCHITECHTURE.md` in `dev-2.x` have resulted in a conflict, could you please also resolve that?

dancesWithCycles commented:
Hi @flaktack ,
Thank you very much for pointing my attention to the `100 character line` formatting rule. Can I assume, this rule is valid for all OTP documentation?

I adjusted the formatting and updated the PR to the upstream branch to incorporate changes and resolve conflicts.

Please feel free to use, study, share and improve.

Cheers!

flaktack commented:
For some reason 52 commits and 164 changed files are showing as part of this PR -- could you please rebase onto `dev-2.x` so that only the relevant commits are shown?

dancesWithCycles commented:
Hi @flaktack ,
that is odd. After you made me aware of the formatting rule and the changes in the upstream branch, I updated my local and remote branch to the upstream branch and finally added my adjustments. 

I do not know or see any way to rebase again. I could close this PR and start a new one from scratch. How about that?

Cheers!

t2gran commented:
In steed of creating a new branch, you may find it easier to cherry-pick your commits:

I suggest you make a new branch or tag from the last commit - this will serve as a backup of you branch.

Then after confirming the new branch is identical to `dancesWithCycles:arch.md`, you can checkout `dancesWithCycles:arch.md` and reset it hard to otp/dev-2.x (`git reset --hard origin/dev-2.x` ). 

Then cherry pick your commits one by one. When done and everything look ok, force push (`push -f`).

dancesWithCycles commented:
Hi @t2gran ,
Thank you so much for your suggestion. It was crystal clear and I followed step-by-step. It turned out that GitHub closed  PR #3451 auto-magically while I was in the middle of the process.

Can you guys point me in the right direction what is the next step for this PR? Is it possible to still use this PR and move forward? Shall I create a new PR with the result of my update?

I thank you very much for your continuous support. Cheers!

t2gran reviewed (approved):
Thank you for improving the text - it is appreciated :-)

dancesWithCycles commented:
Hi @t2gran ,
Thank you for walking me this way. This was my first commit to [OpenTripPlaner](https://github.com/opentripplanner/OpenTripPlanner). Some things are now clear to me that pave the way for future commits. Cheers!

t2gran commented:
No problem, thanks for the contribution. I forgot to comment on the formatting rule for markdown. I am actually not sure what we should do. Having a limit makes it easier to work with (review, merging, and so on), but it seems that some md html translators convert the linebreak into a hard line break, which does not look good. I have not research it, if someone do I am very interested in the answer, so we can make a clear statement in the code-style.
